### PR TITLE
Add support for AArch64

### PR DESCRIPTION
**Pull request by mkurdej**
AArch64 builds need to pass `--build-arg ARCH="aarch64"` and `--build-arg DEVTOOLSET_VERSION=10` when building Docker image.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -4,15 +4,16 @@ ARG PLATFORM="${POLICY}_${ARCH}"
 
 FROM quay.io/pypa/$PLATFORM
 
-ENV PATH="/opt/rh/devtoolset-11/root/usr/bin:$PATH"
+ARG DEVTOOLSET_VERSION="11"
```
Comment (solatis) [inline docker/Dockerfile:7]: ARM needs a different one?

File: quasardb/convert/value.hpp [hunk 2]
```diff
@@ -37,6 +37,7 @@
 #include "../traits.hpp"
 #include "unicode.hpp"
 #include <qdb/ts.h>
+#include <date/date.h> // We cannot use <chrono> until we upgrade to at least GCC11 (ARM).
```
Comment (solatis) [inline quasardb/convert/value.hpp:40]: How do the APIs of date/chrono compare? Looks like they're roughly equivalent?

File: quasardb/convert/value.hpp [hunk 3]
```diff
@@ -364,8 +366,8 @@ struct value_converter<traits::bytestring_dtype, qdb_string_t>
     using char_t = std::string::value_type;
 
     template <concepts::input_range_t<char_t> R>
-    requires(ranges::sized_range<R> && ranges::contiguous_range<R>) inline qdb_string_t operator()(
-        R && x) const
+        requires(ranges::sized_range<R> && ranges::contiguous_range<R>)
+    inline qdb_string_t operator()(R && x) const
```
Comment (solatis) [inline quasardb/convert/value.hpp:370]: Are you using  a different clang-format / linting config? If so, can you somehow sync it into this project?

As far as I'm concerned all these files were already linted using clang-format and these changes are a bit of noise :)

File: docker/Dockerfile (hunk 1 above)
Comment (solatis) [reply docker/Dockerfile:7]: Did you already add a teamcity project for this?

**Review by solatis — COMMENTED** [review]
Looks solid, you had no issues with the concepts / ranges / modern c++ stuff with ARM?

File: docker/Dockerfile (hunk 1 above)
Comment (mkurdej) [reply docker/Dockerfile:7]: Yes, https://teamcity.quasardb.net/buildConfiguration/Api_Python_Release_Linux_Python37AArch64?branch=test&buildTypeTab=overview&mode=builds.

Comment (mkurdej) [reply docker/Dockerfile:7]: amazonlinux image has no GCC 11, so I used 10. This in turn, doesn't have date parts in `<chrono>`, hence I added date library.

File: quasardb/convert/value.hpp (hunk 2 above)
Comment (mkurdej) [reply quasardb/convert/value.hpp:40]: They are pretty much identical. Standard is based on date library.

File: quasardb/convert/value.hpp (hunk 3 above)
Comment (mkurdej) [reply quasardb/convert/value.hpp:370]: I think that it's rather clang-format version that differs. Earlier versions weren't able to format requires clauses.

Comment (mkurdej) [reply quasardb/convert/value.hpp:370]: I'll do formatting changes separately.

Comment (solatis) [reply quasardb/convert/value.hpp]: Ah makes sense then!

**Comment by mkurdej** [comment]
> Looks solid, you had no issues with the concepts / ranges / modern c++ stuff with ARM?

No problem at all. GCC 10 is enough.
Any other remarks?

**Comment by solatis** [comment]
No other comments

**Review by solatis — APPROVED** [review]

**Pull request merged**